homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -147,84 +146,81 @@ def __init__(
         self._to_station = to_station
         self._weekday = weekday
         self._time = departuretime
-        self._state: TrainStop = None
-        self._departure_state = None
+        self._state: TrainStop | None = None
         self._delay_in_minutes = None
         self._timezone = get_time_zone("Europe/Stockholm")
+        self._attr_extra_state_attributes = {
+            ATTR_DEPARTURE_STATE: None,
+            ATTR_CANCELED: None,
+            ATTR_DELAY_TIME: None,
+            ATTR_PLANNED_TIME: None,
+            ATTR_ESTIMATED_TIME: None,
+            ATTR_ACTUAL_TIME: None,
+            ATTR_OTHER_INFORMATION: None,
+            ATTR_DEVIATIONS: None,
+        }
```

Comment:
You don't need to initialise the  state attributes here.
```suggestion
```

Reply:
With the logic I had in update() I had to for consistent always show all attributes regardless if they had info or not. On a former PR on this one I agreed with Martin to not change the logic until the ADR is decided upon.
Regardless I rewrote the logic in update() now so this is gone.

Merry Christmas 🎅